// @prisma/iot-client/wasm.js

/* !!! This is code generated by Prisma. Do not edit directly. !!!
/* eslint-disable */
// biome-ignore-all lint: generated file

Object.defineProperty(exports, "__esModule", { value: true });

const {
  PrismaClientKnownRequestError,
  PrismaClientUnknownRequestError,
  PrismaClientRustPanicError,
  PrismaClientInitializationError,
  PrismaClientValidationError,
  getPrismaClient,
  sqltag,
  empty,
  join,
  raw,
  skip,
  Decimal,
  Debug,
  objectEnumValues,
  makeStrictEnum,
  Extensions,
  warnOnce,
  defineDmmfProperty,
  Public,
  getRuntime,
  createParam,
} = require('./runtime/wasm-engine-edge.js')


const Prisma = {}

exports.Prisma = Prisma
exports.$Enums = {}

/**
 * Prisma Client JS version: 6.19.3
 * Query Engine version: c2990dca591cba766e3b7ef5d9e8a84796e47ab7
 */
Prisma.prismaVersion = {
  client: "6.19.3",
  engine: "c2990dca591cba766e3b7ef5d9e8a84796e47ab7"
}

Prisma.PrismaClientKnownRequestError = PrismaClientKnownRequestError;
Prisma.PrismaClientUnknownRequestError = PrismaClientUnknownRequestError
Prisma.PrismaClientRustPanicError = PrismaClientRustPanicError
Prisma.PrismaClientInitializationError = PrismaClientInitializationError
Prisma.PrismaClientValidationError = PrismaClientValidationError
Prisma.Decimal = Decimal

/**
 * Re-export of sql-template-tag
 */
Prisma.sql = sqltag
Prisma.empty = empty
Prisma.join = join
Prisma.raw = raw
Prisma.validator = Public.validator

/**
* Extensions
*/
Prisma.getExtensionContext = Extensions.getExtensionContext
Prisma.defineExtension = Extensions.defineExtension

/**
 * Shorthand utilities for JSON filtering
 */
Prisma.DbNull = objectEnumValues.instances.DbNull
Prisma.JsonNull = objectEnumValues.instances.JsonNull
Prisma.AnyNull = objectEnumValues.instances.AnyNull

Prisma.NullTypes = {
  DbNull: objectEnumValues.classes.DbNull,
  JsonNull: objectEnumValues.classes.JsonNull,
  AnyNull: objectEnumValues.classes.AnyNull
}





/**
 * Enums
 */
exports.Prisma.TransactionIsolationLevel = makeStrictEnum({
  ReadUncommitted: 'ReadUncommitted',
  ReadCommitted: 'ReadCommitted',
  RepeatableRead: 'RepeatableRead',
  Serializable: 'Serializable'
});

exports.Prisma.NodesScalarFieldEnum = {
  id: 'id',
  node_code: 'node_code',
  name: 'name',
  api_token: 'api_token',
  location_name: 'location_name',
  crop: 'crop',
  active: 'active',
  created_at: 'created_at',
  wifi_ssid: 'wifi_ssid',
  wifi_password: 'wifi_password'
};

exports.Prisma.Sensor_readingsScalarFieldEnum = {
  id: 'id',
  node_id: 'node_id',
  measured_at: 'measured_at',
  air_temp_c: 'air_temp_c',
  air_humidity_pct: 'air_humidity_pct',
  pressure_hpa: 'pressure_hpa',
  leaf_temp_c: 'leaf_temp_c',
  soil_moisture_raw: 'soil_moisture_raw',
  soil_moisture_pct: 'soil_moisture_pct',
  battery_v: 'battery_v',
  rssi_dbm: 'rssi_dbm',
  raw: 'raw',
  created_at: 'created_at'
};

exports.Prisma.SortOrder = {
  asc: 'asc',
  desc: 'desc'
};

exports.Prisma.NullableJsonNullValueInput = {
  DbNull: Prisma.DbNull,
  JsonNull: Prisma.JsonNull
};

exports.Prisma.QueryMode = {
  default: 'default',
  insensitive: 'insensitive'
};

exports.Prisma.NullsOrder = {
  first: 'first',
  last: 'last'
};

exports.Prisma.JsonNullValueFilter = {
  DbNull: Prisma.DbNull,
  JsonNull: Prisma.JsonNull,
  AnyNull: Prisma.AnyNull
};


exports.Prisma.ModelName = {
  nodes: 'nodes',
  sensor_readings: 'sensor_readings'
};
/**
 * Create the Client
 */
const config = {
  "generator": {
    "name": "client",
    "provider": {
      "fromEnvVar": null,
      "value": "prisma-client-js"
    },
    "output": {
      "value": "C:\\Users\\Manuel Q\\Proyectos\\AgroNode\\lib\\node_modules\\@prisma\\iot-client",
      "fromEnvVar": null
    },
    "config": {
      "engineType": "library"
    },
    "binaryTargets": [
      {
        "fromEnvVar": null,
        "value": "windows",
        "native": true
      }
    ],
    "previewFeatures": [],
    "sourceFilePath": "C:\\Users\\Manuel Q\\Proyectos\\AgroNode\\lib\\prisma-iot\\schema.prisma",
    "isCustomOutput": true
  },
  "relativeEnvPaths": {
    "rootEnvPath": null,
    "schemaEnvPath": "../../../../.env"
  },
  "relativePath": "../../../prisma-iot",
  "clientVersion": "6.19.3",
  "engineVersion": "c2990dca591cba766e3b7ef5d9e8a84796e47ab7",
  "datasourceNames": [
    "db"
  ],
  "activeProvider": "postgresql",
  "postinstall": false,
  "inlineDatasources": {
    "db": {
      "url": {
        "fromEnvVar": "DATABASE_URL_IOT",
        "value": null
      }
    }
  },
  "inlineSchema": "generator client {\n  provider = \"prisma-client-js\"\n  output   = \"../node_modules/@prisma/iot-client\"\n}\n\ndatasource db {\n  provider = \"postgresql\"\n  url      = env(\"DATABASE_URL_IOT\")\n}\n\nmodel nodes {\n  id            String   @id @default(dbgenerated(\"gen_random_uuid()\")) @db.Uuid\n  node_code     String   @unique\n  name          String?\n  api_token     String\n  location_name String?\n  crop          String?\n  active        Boolean  @default(true)\n  created_at    DateTime @default(now()) @db.Timestamptz(6)\n  wifi_ssid     String?\n  wifi_password String?\n\n  sensor_readings sensor_readings[]\n\n  @@index([node_code], map: \"idx_nodes_node_code\")\n}\n\nmodel sensor_readings {\n  id                BigInt   @id @default(autoincrement())\n  node_id           String   @db.Uuid\n  measured_at       DateTime @default(now()) @db.Timestamptz(6)\n  air_temp_c        Decimal? @db.Decimal(6, 2)\n  air_humidity_pct  Decimal? @db.Decimal(6, 2)\n  pressure_hpa      Decimal? @db.Decimal(7, 2)\n  leaf_temp_c       Decimal? @db.Decimal(6, 2)\n  soil_moisture_raw Int?\n  soil_moisture_pct Decimal? @db.Decimal(6, 2)\n  battery_v         Decimal? @db.Decimal(5, 2)\n  rssi_dbm          Int?\n  raw               Json?\n  created_at        DateTime @default(now()) @db.Timestamptz(6)\n\n  nodes nodes @relation(fields: [node_id], references: [id], onDelete: Cascade, onUpdate: NoAction)\n\n  @@index([node_id, measured_at(sort: Desc)], map: \"idx_sensor_readings_node_time\")\n}\n",
  "inlineSchemaHash": "9c4228a25d8677d4a82e7e8df409b39b6acadcfae8a6a6601578b1a4a23f044d",
  "copyEngine": true
}
config.dirname = '/'

config.runtimeDataModel = JSON.parse("{\"models\":{\"nodes\":{\"fields\":[{\"name\":\"id\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"node_code\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"name\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"api_token\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"location_name\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"crop\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"active\",\"kind\":\"scalar\",\"type\":\"Boolean\"},{\"name\":\"created_at\",\"kind\":\"scalar\",\"type\":\"DateTime\"},{\"name\":\"wifi_ssid\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"wifi_password\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"sensor_readings\",\"kind\":\"object\",\"type\":\"sensor_readings\",\"relationName\":\"nodesTosensor_readings\"}],\"dbName\":null},\"sensor_readings\":{\"fields\":[{\"name\":\"id\",\"kind\":\"scalar\",\"type\":\"BigInt\"},{\"name\":\"node_id\",\"kind\":\"scalar\",\"type\":\"String\"},{\"name\":\"measured_at\",\"kind\":\"scalar\",\"type\":\"DateTime\"},{\"name\":\"air_temp_c\",\"kind\":\"scalar\",\"type\":\"Decimal\"},{\"name\":\"air_humidity_pct\",\"kind\":\"scalar\",\"type\":\"Decimal\"},{\"name\":\"pressure_hpa\",\"kind\":\"scalar\",\"type\":\"Decimal\"},{\"name\":\"leaf_temp_c\",\"kind\":\"scalar\",\"type\":\"Decimal\"},{\"name\":\"soil_moisture_raw\",\"kind\":\"scalar\",\"type\":\"Int\"},{\"name\":\"soil_moisture_pct\",\"kind\":\"scalar\",\"type\":\"Decimal\"},{\"name\":\"battery_v\",\"kind\":\"scalar\",\"type\":\"Decimal\"},{\"name\":\"rssi_dbm\",\"kind\":\"scalar\",\"type\":\"Int\"},{\"name\":\"raw\",\"kind\":\"scalar\",\"type\":\"Json\"},{\"name\":\"created_at\",\"kind\":\"scalar\",\"type\":\"DateTime\"},{\"name\":\"nodes\",\"kind\":\"object\",\"type\":\"nodes\",\"relationName\":\"nodesTosensor_readings\"}],\"dbName\":null}},\"enums\":{},\"types\":{}}")
defineDmmfProperty(exports.Prisma, config.runtimeDataModel)
config.engineWasm = {
  getRuntime: async () => require('./query_engine_bg.js'),
  getQueryEngineWasmModule: async () => {
    const loader = (await import('#wasm-engine-loader')).default
    const engine = (await loader).default
    return engine
  }
}
config.compilerWasm = undefined

config.injectableEdgeEnv = () => ({
  parsed: {
    DATABASE_URL_IOT: typeof globalThis !== 'undefined' && globalThis['DATABASE_URL_IOT'] || typeof process !== 'undefined' && process.env && process.env.DATABASE_URL_IOT || undefined
  }
})

if (typeof globalThis !== 'undefined' && globalThis['DEBUG'] || typeof process !== 'undefined' && process.env && process.env.DEBUG || undefined) {
  Debug.enable(typeof globalThis !== 'undefined' && globalThis['DEBUG'] || typeof process !== 'undefined' && process.env && process.env.DEBUG || undefined)
}

const PrismaClient = getPrismaClient(config)
exports.PrismaClient = PrismaClient
Object.assign(exports, Prisma)
